Engine/src/Engine/ResourceManagement/ShaderNorms.h

```diff
@@ -15,24 +16,26 @@ namespace Engine
 	class ShaderNorms
 	{
 	public:
-		//shader tokens to access map of shaders
-		static enum ShaderName
-		{
-			SN_TEXTURE_FILL,
-			SN_COLOR_FILL,
-			SN_TEXT_FILL,
-			SN_GRADIENT_FILL
-		};
-		static const ShaderName DEFAULT_SHADER_NAME;
+		//any time-related uniform in the advanced shader
+		const static inline float TIME_SPEEDUP_MOD = 3.f;
 
 		//requires the stride of the default shader's vertices component
 		~ShaderNorms();
 		ShaderNorms(ShaderNorms& other) = delete;
 		void operator=(const ShaderNorms&) = delete;
 		static ShaderNorms* getInstance();
 
+		//sets the uniform value of a float
+		void setUniformValuef(int& bind, std::string uniformName, float value);
+
+		//sets the uniform value of an int
+		void setUniformValuei(int& bind, std::string uniformName, int value);
+
+		//sets the uniform value of a bool
+		void setUniformValueb(int& bind, std::string uniformName, bool value);
+
 		//load shader according to vertices component (given shader is not already loaded)
-		void update(int advancedShaderBind, int stride, int& textureCoordinates,
+		void update(double time, int advancedShaderBind, int stride, int& textureCoordinates,
```

Comment:
we can use DeltaTime here to be more consistent with our engine

Reply:
Done